Kadena Address Changes

mankyKitty opened the pull request:
Change the checksum to pure haskell CRC32 using `crc` package.

- Remove network name from the address.
- Add a flag to indicate if the account has been created yet, defaults to no.
- Remove `uri-bytestring` dependency.
- Update test.

Commit:
Kadena Address Updates

Change the checksum to pure haskell CRC32 using `crc` package.

Remove network name from the address.

Add a flag to indicate if the account has been created yet, defaults to no.

Remove `uri-bytestring` dependency.

update test.

alexfmpe commented on frontend/frontend.cabal:
```diff
@@ -79,7 +79,7 @@ library
                , unordered-containers
                , kadena-signing-api
                , base64-bytestring
-               , uri-bytestring
```
Can probably remove `hashing` too

Commit:
Remove 'encoded' field from Address

The encoded field introduced a possibility for inconsistent or corrupted
address information, this has been refactored to just be a function.

alexfmpe commented on frontend/src/Frontend/KadenaAddress.hs:
```diff
@@ -101,21 +102,31 @@ data KadenaAddressError
+  | KadenaAddressPrefix_None
+  deriving (Show, Eq)
+
+data KadenaAddress_AccountCreated
```
Don't think we actually use `_` outside of field/constructor prefixes, convenient as it would be. Might mess with lens-making?

Commit:
Change address encoding based on created status.

Accounts that are not yet created will have their address displayed as the account name,
chain id, and the checksum.

Accounts that _are_ created will have their address displayed as the account name, chain
id, then the base 64 encoded collection of that information plus a check sum.

Commit:
Minor de-duplication

alexfmpe commented on frontend/src/Frontend/KadenaAddress.hs:
```diff
@@ -131,57 +141,65 @@ bsshow = fromString . show
+    <> cons humanReadableDelimiter (case _kadenaAddress_accountCreated ka of
+                                      AccountCreated_Yes -> encoded
+                                      AccountCreated_No -> checksum
+                                   )
```
```haskell
intersperse humanReadableDelimiter
[ name
, cid
, case _kadenaAddress_accountCreated ka of
    AccountCreated_Yes -> encoded
    AccountCreated_No -> checksum
]

Commit:
Improvement from PR feedback

Commit:
Update docs for kadenaaddress
